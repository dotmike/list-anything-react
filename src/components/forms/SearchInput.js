import React, { Component } from "react";
import PropTypes from "prop-types";
import { Form, Input } from "semantic-ui-react";

class SearchInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: false
    };
    this.icon = {
      name: "search",
      link: true,
      onClick: this.onSubmit
    };
  }

  onSubmit = () => {
    const $input = this.getInput();
    const query = $input.value;
    if (query) {
      this.props.onSearch(query);
      this.setState({ error: false });
    } else this.setState({ error: true });
    $input.value = "";
    $input.blur();
  };

  getInput = () =>
    document.querySelector(`#${this.props.id} input[name="query"]`);

  render() {
    const { error } = this.state;
    return (
      <Form id={this.props.id} onSubmit={this.onSubmit}>
        <Input
          icon={this.icon}
          placeholder="Search..."
          size="mini"
          error={error}
          name="query"
        />
      </Form>
    );
  }
}

SearchInput.propTypes = {
  // ownProps
  id: PropTypes.string,
  onSearch: PropTypes.func.isRequired
};

SearchInput.defaultProps = {
  id: `seach-form-${Math.floor(Math.random() * 100000)}`
};

export default SearchInput;
