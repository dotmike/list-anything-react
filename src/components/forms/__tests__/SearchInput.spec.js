import React from "react";
import { shallow } from "enzyme";

import SearchInput from "../SearchInput";

const defaultProps = {
  id: "seach-form-1",
  onSearch: () => {}
};

describe("SearchInput", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<SearchInput {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
