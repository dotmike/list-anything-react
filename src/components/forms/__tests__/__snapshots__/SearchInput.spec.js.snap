// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders correctly 1`] = `
<Form
  as="form"
  id="seach-form-1"
  onSubmit={[Function]}
>
  <Input
    error={false}
    icon={
      Object {
        "link": true,
        "name": "search",
        "onClick": [Function],
      }
    }
    name="query"
    placeholder="Search..."
    size="mini"
    type="text"
  />
</Form>
`;
